### PR TITLE
Fix https://github.com/webpack/webpack/issues/1645

### PR DESCRIPTION
**Pull request by BlackGlory**
Make plugins filter false value.

### CODE REVIEW

**Comment by loganfsmyth** [comment]
Babel's generally very strict about arguments conforming to the expected inputs. I think it would be more consistent to throw an error message about unexpected blank plugins than to silently filter them out.

In your specific case, this is the primary reason that trailing commas introduce less confusion.

**Comment by thejameskyle** [comment]
An error is definitely the best option here.

**Comment by hzoo** [comment]
@BlackGlory, can you modify your PR to error on falsy plugin values?

File: packages/babel-core/src/transformation/file/options/option-manager.js [hunk 1]
```diff
@@ -117,10 +117,14 @@ export default class OptionManager {
       let plugin, options;
 
       // destructure plugins
-      if (Array.isArray(val)) {
-        [plugin, options] = val;
+      if (val) {
```
Comment (ForbesLindesay) [inline packages/babel-core/src/transformation/file/options/option-manager.js]: Should we also check `typeof val`?  I think it should always be either "Array", `'string'` or `'function'`.

**Comment by jmm** [comment]
There should be a test added for this.

**Comment by jmm** [comment]
1. If the goal here is to alert the user more specifically and as early as possible, then we need a better error message. We should probably just say something like:

  > Falsy value found in `plugins`

  @BlackGlory note: you don't need to worry about `babel-messages` / `messages.get()`, you can just inline the error message here.

2. I think `TypeError` is more appropriate here.

3. We should get rid of this unnecessary `else` and nesting and just do:

  ```js
  if (!val) // throw
  // carry on
  ```

  (For that matter there's another unecessary `else` and variable here we can get rid of later -- not something you need to worry about @BlackGlory.)

4. Please squash your commits and just force push the new one to your fork.

**Comment by BlackGlory** [comment]
@jmm Thank you, I just do not have time to pay attention to the PR, I will create a new one later.

**Pull request closed without merging**